rust/Cargo.toml

```diff
@@ -24,6 +24,9 @@ chrono = "0"
 uuid = { version = "1", features = ["serde", "v4"] }
 lazy_static = "1"
 percent-encoding = "2"
+hyper = "0.14.19"
+hyper-rustls = {  version = "0.23.0", features = ["http2"] } 
+hyper-proxy = {  version = "0.9.1", default-features = false, features = ["rustls"] }
```

Comment:
could we make these conditional, and hide behind the S3 feature gate.

Reply:
Yes, I'm not sure if I need to hide it between both s3 and s3-rustls ? Because hyper-rustls uses rustls but the feature is needed even if you don't use s3-rusttls impl